File: AddAPearl.Web/src/app/shared/models/customer.ts
import {
  IAddress,
  ICompany,
} from './index';

/* Defines the customer entity */
export interface ICustomer {
  customerId: number;
  firstName: string;
  lastName: string;
  phoneNumber: string;
  birthDay: string;
  email: string;
  addressId: number;
  address: IAddress;
  companyId: number;
  company: ICompany;
}

export class Customer implements ICustomer {
  constructor(
    public customerId: number,
    public firstName: string,
    public lastName: string,
    public phoneNumber: string,
    public birthDay: string,
    public email: string,
    public addressId: number,
    public address: IAddress,
    public companyId: number,
    public company: ICompany,
  ) {
    // Any other stuff to do during construction...
  }
}

File: src/AddAPearl.Domain/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using AddAPearl.Core;

namespace AddAPearl.Domain
{
    public class Customer : ICustomer
    {
        public int CustomerId { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string PhoneNumber { get; set; }

        public DateTime BirthDay { get; set; }

        public string Email { get; set; }

        public int CompanyId { get; set; }

        public ICompany Company { get; set; }

        public int AddressId { get; set; }

        public IAddress Address { get; set; }
    }
}
